addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/handler/BridgeHandler.java

```diff
@@ -0,0 +1,512 @@
+                            i = i + 1;
+                        } while (i < o.length);
+                    }
+                } else {
+                    // no OpeningTimes found, assuming WholeDay open!
+                    foundIt = true;
+                    logger.debug("No OpeningTimes are found, assuming WholeDay.");
+                    // "start" and "ende" are set manually!
+                    start = "00:00";
+                    ende = "23:59";
+                }
+                LocalTime tempopening = LocalTime.parse(start);
+                LocalTime tempclosing = LocalTime.parse(ende);
+                DateTime opening = tempopening.toDateTimeToday();
+                DateTime closing = tempclosing.toDateTimeToday();
+                if (opening.isEqual(closing)) {
+                    closing = opening.plusDays(1);
+                } else {
+                    // Tankerkoenig.de does update the status "open" every 4 minutes
+                    // due to this the status "open" could be sent up to 4 minutes after the published opening time
+                    // therefore the first update is called 4 minutes after opening time!
+                    opening = opening.plusMinutes(4);
+                }
+                if ((now.isAfter(opening) & (now.isBefore(closing)))) {
+                    // logger.debug("Opening: {}, Closing: {}, now: {}", opening, closing, now);
+                    logger.debug("Now is within opening times for today.");
+                    if (sb.toString().equals("")) {
+                        sb.append(tkh.getLocationID());
+                    } else {
+                        sb.append("," + tkh.getLocationID());
+                    }
+                }
+            }
+            return sb.toString();
+        } catch (Exception e) {
+            logger.debug("Exception in BridgeHandler: {}", e);
+            return null;
+        }
+    }
+
+    public String getApiKey() {
+        return apiKey;
+    }
+
+    public void setApiKey(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public int getRefreshInterval() {
+        return refreshInterval;
+    }
+
+    public void setRefreshInterval(int refreshInterval) {
+        this.refreshInterval = refreshInterval;
+    }
+
+    public TankerkoenigListResult getTankerkoenigListResult() {
+        return tankerkoenigListResult;
+    }
+
+    public void setTankerkoenigListResult(TankerkoenigListResult tankerkoenigListResult) {
+        this.tankerkoenigListResult = tankerkoenigListResult;
+    }
+
+    public boolean isSetupMode() {
+        return setupMode;
+    }
+
+    public void setSetupMode(boolean setupMode) {
+        this.setupMode = setupMode;
+    }
+
+    public boolean isUseOpeningTime() {
+        return use_OpeningTime;
+    }
+
+    public void setUseOpeningTime(boolean use_OpeningTime) {
+        this.use_OpeningTime = use_OpeningTime;
+    }
+
+    public boolean isHoliday(DateTime now) {
```

Comment:
Sure I can do that, however I copied that code from the rule examples and I'm not sure whom to put into the signoff (which dolic and me have forgotten so far).
The other constraint from me is that I'm not sure on the future usage of that part. I incooperated it for the moment with all holidays that are valid all over Germany (Bundesweite Feiertage). I do forsee that the need migth arise to make a list of selectable holydays. Whre should that be selected? On PaperUI or in a .cfg file? Or am I overcomplicating that, and nobody would be using the feature OpeningTimes at all?